core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -117,10 +127,11 @@ class ReplicaFetcherThread(name: String,
 
     // For the follower replica, we do not need to keep its segment base offset and physical position.
     // These values will be computed upon becoming leader or handling a preferred read replica fetch.
-    val followerHighWatermark = log.updateHighWatermark(partitionData.highWatermark)
+    val highWatermarkUpdate = log.updateHighWatermark(partitionData.highWatermark)
+    info(s"follower high watermark: ${highWatermarkUpdate.highWatermark}")
```

Comment:
Should it be trace?  This could flood the info log. Also looks like we already log this info a few lines below.